### PR TITLE
CORE-7276 argumentless sign

### PR DESCRIPTION
**Pull request by vlajos**
CORE-7276 Remove toSignedTransaction(PublicKey)s from TxBuilders

Runtime-os: https://github.com/corda/corda-runtime-os/pull/2981

### CODE REVIEW

**Review by MrMatthewLayton — APPROVED** [review]
LGTM

**Review by MrMatthewLayton — DISMISSED** [review]

**Review by MrMatthewLayton — APPROVED** [review]
LGTM

**Review by lankydan — APPROVED** [review]

File: ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt [hunk 1]
```diff
@@ -38,24 +38,8 @@ interface ConsensualTransactionBuilder {
      *
      * @throws IllegalStateException when called a second time on the same object to prevent
      *      unintentional duplicate transactions.
+     * @throws CordaRuntimeException if none of the required keys are available to sign the transaction.
```
Comment (kyriathar) [inline ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt:41]: Feels like this should be of a more specific exception type? (Also doesn't feel like a runtime one)

**Review by kyriathar — APPROVED** [review]
Only a minor comment otherwise LGTM

File: ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt (hunk 1 above)
Comment (vlajos) [reply ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt:41]: I am a bit uncertain about this one.
We should not have SignedTxs without signatures.
But the TransactionSIgnature service should not fail if there are no keys. That can just return an empty list.
So I ended up testing if any signatures were generated in the SignedTxFactories and throwing this exception if there were none.

**Pull request merged**